src/fileNamer.js

```diff
@@ -151,3 +151,139 @@ function mapValue(map, value) {
     return `-${map[value]}`
   }
 }
+
+function evaluateNewLabel(documents, L, LWP) {
+  let failed = 0
+  for (let doc of documents) {
+    doc.category = evalCategory(doc)
+    if (doc.category === undefined) {
+      if (LWP) {
+        log('info', `New label: ${doc.label}`)
+      }
+      log('warn', 'Category Unknown')
+      failed++
+      continue
+      //throw 'Category Unknown'
+    }
+    doc.type = evalType(doc)
+    if (doc.type === undefined) {
+      if (LWP) {
+        log('info', `New label: ${doc.label}`)
+      }
+      log('warn', 'Type Unknown')
+      failed++
+      continue
+      //throw 'Type Unknown'
+    }
+    doc.form = evalForm(doc, doc.year, doc.category)
```

Comment:
Since evalForm is already receiving `doc`, I do not think it is necessary to pass `year` and `category`.

Reply:
Yes, thank's, result of first test implementation, will change